src/v/cloud_storage/async_manifest_view.cc

```diff
@@ -1005,15 +1007,31 @@ async_manifest_view::compute_retention(
           "Found offset {} to advance start offset to",
           result.offset);
     }
+    if (pinned_offset && result.offset - result.delta > *pinned_offset) {
```

Comment:
It looks like this computation could be performed outside of this method.

Reply:
Hm I didn't have a clear sense of what you were thinking -- mind clarifying?